src/detection/wm/wm_linux.c

```diff
@@ -26,15 +26,15 @@ static bool extractHyprlandVersion(const char* line, FF_MAYBE_UNUSED uint32_t le
 static const char* getHyprland(FFstrbuf* result)
 {
     FF_STRBUF_AUTO_DESTROY path = ffStrbufCreate();
-    const char* error = ffFindExecutableInPath("Hyprland", &path);
```

Comment:
The version string should be extracted from Hyprland binary file directly by `ffFindExecutableInPath`, which is equivalent to `strings (which Hyprland) | rg 'v\d+\.\d+\.\d+'`. This is almost zero cost because it doesn't spawn processes at all.

If it doesn't work, fastfetch is either built without libelf support, or fastfetch fails to load `libelf.so`. You should check it